src/screens/modals/Player/components/Timer.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import DateTimePickerModal from 'react-native-modal-datetime-picker';
+import styled from 'styled-components/native';
+import { THEME_COLOR } from '@/CONSTANTS';
+import { useDispatch } from 'react-redux';
+import { setDateTime } from '@/store/settings/actions';
+import { useTypedSelector } from '@/store';
+import TimerIcon from '@/assets/icons/timer-icon.svg';
+
+const Container = styled.View`
+    align-item: left;
+    margin-top: 60px;
+`;
+
+const View = styled.View`
+    display: flex;
+    flex-direction: row;
+    gap: 4px;
+`;
+
+const Text = styled.Text`
+    font-size: 10px;
+`;
+
+export default function Timer() {
+    const [showPicker, setShowPicker] = useState<boolean>(false);
+    const { remainingSleepTime } = useTypedSelector(state => state.settings);
```

Comment:
Rather than retrieving the remaining time from the store, you should calculate it in this component. You can do that with some clever intervals and a state to keep the remaining time. 

```tsx
function Timer() {
    const [remainingTime, setRemainingTime] = useState<null | string>(null);
    const { timerDate } = useTypedSelector((state) => state.settings);

    useEffect(() => {
        if (!timerDate) {
            setRemainingTime(null);
            return;
        }

        const interval = setInterval(() => {
            const diffMs = timerDate.getTime() - new Date().getTime();
            setRemainingTime(ticksToDuration(diffMs));
        }, 1000);

        return () => clearInterval(interval);
    }, [timerDate]);

    return remainingTime;
}
```

Find the `ticksToDuration` function in `./src/utility/ticksToDuration.ts`